src/Microsoft.Health.Fhir.Liquid.Converter/Filters/SectionFilters.cs

```diff
@@ -0,0 +1,138 @@
+﻿// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using System;
+using System.Collections.Generic;
+using System.Text.RegularExpressions;
+using DotLiquid;
+
+namespace Microsoft.Health.Fhir.Liquid.Converter
+{
+    /// <summary>
+    /// Filters for conversion
+    /// </summary>
+    public partial class Filters
+    {
+        private static readonly Regex NormalizeSectionNameRegex = new Regex("[^A-Za-z0-9]");
+
+        public static IDictionary<string, object> GetFirstCdaSections(Hash data, string sectionNameContent)
+        {
+            var result = new Dictionary<string, object>();
+            var sectionNames = sectionNameContent.Split("|", StringSplitOptions.RemoveEmptyEntries);
+            var components = GetComponents(data);
+
+            if (components == null)
+            {
+                return result;
+            }
+
+            foreach (var sectionName in sectionNames)
+            {
+                foreach (var component in components)
+                {
+                    if (component is Dictionary<string, object> componentDict &&
+                        componentDict.GetValueOrDefault("section") is Dictionary<string, object> sectionDict &&
+                        sectionDict.GetValueOrDefault("title") is Dictionary<string, object> titleDict &&
+                        titleDict.GetValueOrDefault("_") is string titleString &&
+                        titleString.Contains(sectionName, StringComparison.InvariantCultureIgnoreCase))
+                    {
+                        result[NormalizeSectionName(sectionName)] = sectionDict;
+                        break;
+                    }
+                }
+            }
+
+            return result;
+        }
+
+        public static IDictionary<string, object> GetCdaSectionLists(Hash data, string sectionNameContent)
```

Comment:
Can we leverage `GetCdaSectionLists` in `GetFirstCdaSections` since they have many common parts?
Say `GetFirstCdaSections => GetCdaSectionLists()[normalizedSectionName].First()`.

Reply:
Yes! I've updated `GetFirstCdaSections` using `GetCdaSectionLists`.